zerver/lib/actions.py

```diff
@@ -3636,11 +3647,18 @@ def do_update_pointer(user_profile: UserProfile, client: Client,
 def do_mark_all_as_read(user_profile: UserProfile, client: Client) -> int:
     log_statsd_event('bankruptcy')
 
-    msgs = UserMessage.objects.filter(
+    message_ids_to_not_mark_as_read = UserMessage.objects.filter(user_profile=user_profile
+    ).extra(
+        where=[UserMessage.where_unread()]
+    ).order_by('-id').values_list('id', flat=True)[:ONBOARDING_NUM_MESSAGES]
+
+    msgs = UserMessage.objects.exclude(
+        id__in=message_ids_to_not_mark_as_read
+    ).filter(
```

Comment:
@modestfool why did you change this function?  I don't understand why this would be relevant to what you're doing.